Cabal/doc/developing-packages.markdown

```diff
@@ -1476,13 +1482,45 @@ for these fields.
     `install-includes` is typically used to name header files that
     contain prototypes for foreign imports used in Haskell code in this
     package, for which the C implementations are also provided with the
-    package.  Note that to include them when compiling the package
-    itself, they need to be listed in the `includes:` field as well.
+    package. For example, here is a `.cabal` file for a hypothetical
+    `bindings-clib` package that bundles the C source code for `clib`:
+
+    ~~~~~~~~~~~~~~~~
+    include-dirs:     cbits
+    c-sources:        clib.c
+    install-includes: clib.h
+    ~~~~~~~~~~~~~~~~
+
+    Now any package that depends (directly or transitively) on the
+    `bindings-clib` library can include `clib.h` without needing to list it
+    explicitly in an `include` field.
+
+    Note that in order for files listed in `install-includes` to be usable
+    when compiling the package itself, they need to be listed in the
+    `includes:` field as well.
 
 `include-dirs:` _directory list_
 :   A list of directories to search for header files, when preprocessing
     with `c2hs`, `hsc2hs`, `cpphs` or the C preprocessor, and
-    also when compiling via C.
+    also when compiling via C. Directories can be absolute paths (e.g., for
+    system directories) or paths that are relative to the top of the source
+    tree.
+
+    Cabal looks in these directories when attempting to locate files listed in
+    `includes` and `install-includes`. As a result, the following:
+
+    ~~~~~~~~~~~~~~~~
+    includes:         foo/bar.h
+    install-includes: foo/baz.h
+    ~~~~~~~~~~~~~~~~
+
+    is equivalent to the following:
+
+    ~~~~~~~~~~~~~~~~
+    include-dirs:     foo
+    includes:         bar.h
+    install-includes: baz.h
+    ~~~~~~~~~~~~~~~~
```

Comment:
No, I don't think they're equivalent. In particular, if you say `includes: foo/bar.h`, then in your source files you have to say `#include <foo/bar.h>`.